File: VB/MultiDateEditCalendar.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.ViewInfo
Imports System.Drawing

Namespace MultiDateEditor
	' my multi date edit calendar class
	Public Class MultiDateEditCalendar
		Inherits DateEditCalendar
		'constructor
		Public Sub New(ByVal item As RepositoryItemDateEdit, ByVal editValue As Object)
			MyBase.New(item, editValue)
		End Sub

		' new property
		Protected Friend ReadOnly Property CalendarsRecr() As Rectangle
			Get
				Return MyBase.CalendarsRect
			End Get
		End Property

		Protected Overrides Sub ChangeEditDateCore(ByVal yearOffset As Integer, ByVal monthOffset As Integer, ByVal daysOffset As Integer)
			Dim iCalendarsCount As Integer = CInt(Fix((TryCast(Properties, RepositoryItemMultiDateEdit)).MultiEditorType))
			MyBase.ChangeEditDateCore(yearOffset, iCalendarsCount * monthOffset, daysOffset)
		End Sub

		Protected Overrides Function CreateInfoArgs() As DateEditInfoArgs
			Return New MultiDateEditInfoArgs(Me)
		End Function
	End Class
End Namespace

File: CS/MultiDateEditCalendar.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.ViewInfo;
using System.Drawing;

namespace MultiDateEditor
{
    // my multi date edit calendar class
    public class MultiDateEditCalendar : DateEditCalendar
    {
        //constructor
        public MultiDateEditCalendar(RepositoryItemDateEdit item, object editValue) : base(item, editValue) { }

        // new property
        protected internal Rectangle CalendarsRecr { get { return base.CalendarsRect; } }

        protected override void ChangeEditDateCore(int yearOffset, int monthOffset, int daysOffset)
        {
            int iCalendarsCount = (int)(Properties as RepositoryItemMultiDateEdit).MultiEditorType;
            base.ChangeEditDateCore(yearOffset, iCalendarsCount * monthOffset, daysOffset);
        }

        protected override DateEditInfoArgs CreateInfoArgs()
        {
            return new MultiDateEditInfoArgs(this);
        }
    }
}
